datadistillrSDK/pythonSDK.py

```diff
@@ -0,0 +1,139 @@
+    organizations_list = organizations = base_url + "organization"
+    logout_page = base_url + 'logout'
+
+    def __init__(self, email, password):
+        """
+        The constructor for Datadistillr class. Creates a session.
+
+        Parameters:
+            email (string): The email linked to Datadistillr account.
+            password (string): The password linked to Datadistillr account.
+        """
+
+        requests.packages.urllib3.disable_warnings(InsecureRequestWarning)
+        # stores cookies, so you can make requests without multiple logins (pass around cookie)
+        self.session = requests.Session()
+        self.email = email
+        self.password = password
+        self.login_resp_json = self.login(self.email, self.password)
+        self.is_logged_in = self.login_resp_json["loggedIn"]
+
+    def login(self, email, password):
+        """
+        Login and authenticate to DataDistillr.
+
+        Parameters:
+            email (string): The email linked to account.
+            password (string): The password linked to account.
+
+        Returns:
+            json: A json containing account details and login status.
+        """
+
+        user_info = {
+            "email": email,
+            "password": password,
+            "invitations": {
+                "organizationInvitationToken": None,
+                "projectInvitationToken": None,
+                "teamInvitationToken": None}
+        }
+
+        login_response = self.session.post(url=self.login_page, json=user_info, verify=False)
+        login_resp_json = login_response.json()
+        return login_resp_json
+
+    def logout(self):
+        """
+        Log user out of DataDistillr account.
+
+        Returns:
+            json: A json containing account details and login status.
+        """
+
+        logout_response = self.session.get(url=self.logout_page, verify=False)
+        logout_resp_json = logout_response.json()
+        self.is_logged_in = logout_resp_json["loggedIn"]
+        return logout_resp_json
+
+    def get_projects(self):
+        """
+        Gets all projects in DataDistillr account.
+
+        Returns:
+            list<project>: A list of project objects
+        """
+
+        if not self.is_logged_in:
+            return "login is incorrect"
+
+        # URL to get project needs active org token, URL format may change
+        org_token = self.login_resp_json["activeOrganization"]["token"]
+        projects_page = self.base_url + "organization/" + str(org_token) + "/projects"
+        projects_response = self.session.get(url=projects_page, verify=False)
+
+        # Converts response to JSON
+        proj_resp_json = projects_response.json()
+
+        # Gets the projects list
+        proj_list = proj_resp_json["projects"]
+        proj_object_list = []
+
+        for i in range(len(proj_list)):
+            # create new project object with json
+            proj_object = project(proj_list[i], self.session)
```

Comment:
Do you need the `self` here?

Reply:
yes, it returns a variable not found error when I remove this.